resource-aggregate/cqrs/eventstore/mongodb/load.go

```diff
@@ -309,26 +322,134 @@ func snapshotQueriesToMongoQuery(groupID string, queries []eventstore.SnapshotQu
 	})
 	if len(queries) == 0 {
 		opts.SetHint(groupIDQueryIndex)
-		return bson.D{
-			{Key: groupIDKey, Value: groupID}, {Key: isActiveKey, Value: true},
-		}, opts
+		return []mongoQuery{
+			{
+				filter: bson.D{
+					{Key: groupIDKey, Value: groupID}, {Key: isActiveKey, Value: true},
+				},
+				options: opts,
+			},
+		}
 	}
 
-	opts.SetHint(groupIDaggregateIDQueryIndex)
-	orQueries := make([]bson.D, 0, 32)
+	optsTypes := options.Find()
+	// create a copy of the options
+	*optsTypes = *opts
+	resourceQueries := make([]bson.D, 0, 32)
+	typeQueries := make([]bson.D, 0, 32)
 	for _, q := range queries {
+		if q.AggregateID == "" && len(q.Types) == 0 {
+			opts.SetHint(groupIDQueryIndex)
+			return []mongoQuery{
+				{
+					filter: bson.D{
+						{Key: groupIDKey, Value: groupID}, {Key: isActiveKey, Value: true},
+					},
+					options: opts,
+				},
+			}
+		}
+		if q.AggregateID == "" && len(q.Types) > 0 {
+			optsTypes.SetHint(groupIDTypesQueryIndex)
+			typeQueries = append(typeQueries, bson.D{{Key: groupIDKey, Value: groupID}, {Key: typesKey, Value: bson.M{"$all": q.Types}}, {Key: isActiveKey, Value: true}})
```

Comment:
What is the expected behavior here? Do we want exact matches or not? Becasue `{"$all": q.Types}` means that it must contain all values from the `q.Types`, however it can also contain others.

So if `q.Types = []string["a", "b"]` it will match documents contain just `["a", "b"]`, but also documents that have `["a", "b", "c"]`. Is that expected?